src/main/java/com/google/api/codegen/DocConfig.java

```diff
@@ -0,0 +1,185 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen;
+
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.TypeRef;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import javax.annotation.Nullable;
+
+public abstract class DocConfig {
+  public abstract String getApiName();
+
+  public abstract String getMethodName();
+
+  public abstract String getReturnType();
+
+  // FIXME: is this Java-specific?
```

Comment:
This is a deliberate violation to ensure that I address all FIXME comments prior to merging to master. Only TODOs will be part of that merge.